runtime/memory.c

```diff
@@ -533,59 +532,16 @@ Caml_inline value caml_alloc_shr_aux (mlsize_t wosize, tag_t tag, int track,
   return Val_hp (hp);
 }
 
-#ifdef WITH_PROFINFO
-
-/* Use this to debug problems with macros... */
-#define NO_PROFINFO 0xff
-
-CAMLexport value caml_alloc_shr_with_profinfo (mlsize_t wosize, tag_t tag,
-                                               intnat profinfo)
-{
-  return caml_alloc_shr_aux(wosize, tag, 1, 1, profinfo);
-}
-
-CAMLexport value caml_alloc_shr_for_minor_gc (mlsize_t wosize,
-                                              tag_t tag, header_t old_header)
-{
-  return caml_alloc_shr_aux (wosize, tag, 0, 1, Profinfo_hd(old_header));
-}
-
-#else
-#define NO_PROFINFO 0
-
-CAMLexport value caml_alloc_shr_for_minor_gc (mlsize_t wosize,
-                                              tag_t tag, header_t old_header)
```

Comment:
`caml_alloc_shr_for_minor_gc` is still necessary - it calls `caml_alloc_shr_aux` with `track = 0`, to prevent minor allocations being resampled by Memprof when they're promoted. (This may be why the statmemprof tests are failing)

Reply:
Thanks for the confirmation, I was looking exactly now!